lib/webpush/request.rb

```diff
@@ -36,52 +37,110 @@ def headers
       headers["Content-Type"] = "application/octet-stream"
       headers["Ttl"]          = ttl
 
-      if encrypted_payload?
+      if @payload.has_key?(:server_public_key)
         headers["Content-Encoding"] = "aesgcm"
         headers["Encryption"] = "salt=#{salt_param}"
         headers["Crypto-Key"] = "dh=#{dh_param}"
       end
 
-      headers["Authorization"] = "key=#{api_key}" if api_key?
+      if api_key?
```

Comment:
@rossta Thank you for your great PR! 😀
I think that it cannot be sent to the FF endpoint is not a VAPID.
I think that judging by the vapid options.
What do you think?